backend/app/api/endpoints/users.py

```
@@ -27,6 +27,35 @@
+# ==================== Feature Flags ====================
+
+
+class FeatureFlags(BaseModel):
+    """System-level feature flags for the frontend"""
+
+    memory_enabled: bool = False  # Whether long-term memory service is available
+
+
+@router.get("/features", response_model=FeatureFlags)
+async def get_feature_flags(
+    _current_user: User = Depends(security.get_current_user),  # noqa: ARG001
```

Ruff (0.14.14)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)